source/opcode.cpp

```diff
@@ -417,3 +417,22 @@ bool spvOpcodeIsBlockTerminator(SpvOp opcode) {
   return spvOpcodeIsBranch(opcode) || spvOpcodeIsReturn(opcode) ||
          opcode == SpvOpKill || opcode == SpvOpUnreachable;
 }
+bool spvOpcodeIsOpaqueType(SpvOp opcode) {
+  switch(opcode) {
+    case SpvOpTypeImage:
+    case SpvOpTypeSampler:
+    case SpvOpTypeSampledImage:
+    case SpvOpTypeOpaque:
+    case SpvOpTypeEvent:
+    case SpvOpTypeDeviceEvent:
+    case SpvOpTypeReserveId:
+    case SpvOpTypeQueue:
+    case SpvOpTypePipe:
+    case SpvOpTypeForwardPointer:
```

Comment:
I expected that OpTypeForwardPointer should be the same as OpTypePointer, and should not be on this list.  But I see the SPIR-V spec says it's an opaque type.
But OpTypeForwardPointer doesn't even have a result id....   :-)

I guess this is fine.  You can't do better.

Reply:
Okay, I'll leave it as is.